lib/Elasticsearch/QueryConditionGenerator.php

```diff
@@ -31,6 +31,7 @@
     public const QUERY_BOOL = 'bool';
     public const QUERY_IDS = 'ids';
     public const QUERY_NESTED = 'nested';
+    public const QUERY_HAS_CHILD = 'has_child';
```

Comment:
In general are these constants needed? These are not magical numbers, and the name basically already describes the value.

Secondly, the Symfony DI Component removed the "direct" usage of constants as they are slower in some cases 🤔 I tried to benchmark this, but I couldn't think of a good representative condition.